File: app/src/main/java/com/csstudios/developers/theultimatestressreliever/AdviceActivity.java
package com.csstudios.developers.theultimatestressreliever;

import android.content.Intent;
import android.media.MediaPlayer;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.Spinner;
import java.util.Arrays;
import java.lang.String;

public class AdviceActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_advice);

        // Media players for advice (One for each recording)
        final MediaPlayer adviceSound = MediaPlayer.create(this, R.raw.advice1);
        final MediaPlayer adviceSound2 = MediaPlayer.create(this, R.raw.advice2);
        final MediaPlayer adviceSound3 = MediaPlayer.create(this, R.raw.advice3);
        final MediaPlayer adviceSound4 = MediaPlayer.create(this, R.raw.advice4);
        final MediaPlayer adviceSound5 = MediaPlayer.create(this, R.raw.advice5);
        final MediaPlayer adviceSound6 = MediaPlayer.create(this, R.raw.advice6);

        final Button adviceButton = (Button) findViewById(R.id.adviceButton);

        //Select advice with spinner
        Spinner spinnerAdvice = (Spinner) findViewById(R.id.spinnerAdvice);
        String[] adviceArray = getResources().getStringArray(R.array.advice);

        ArrayAdapter<String> adapter = new ArrayAdapter<String>(
                this, android.R.layout.simple_list_item_1, Arrays.asList(adviceArray));

        spinnerAdvice.setAdapter(adapter);
        spinnerAdvice.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                final String selectedAdvice = parent.getSelectedItem().toString();
                adviceButton.setOnClickListener(new View.OnClickListener() {
                    @Override
                    // Detection of Spinner Selection
                    public void onClick(View v) {
                        // Advice selected, then plays sound and stops all other sounds
                        switch (selectedAdvice) {
                            case "Fun Fact": adviceSound.start();
                                break;
                            case "Random Advice": adviceSound2.start();
                                break;
                            case "Life-changing Advice": adviceSound3.start();
                                break;
                            case "Another Fun Fact": adviceSound4.start();
                                break;
                            case "Did You Know?": adviceSound5.start();
                                break;
                            case "Just One More Thing...": adviceSound6.start();
                                break;
                        }

                    }
                });
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });
        // Switch Activities
        ImageButton buttonSwitch = (ImageButton) findViewById(R.id.btnSwitch);
        buttonSwitch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent mainScreen = new Intent(v.getContext(), MainActivity.class);
                startActivity(mainScreen);
            }
        });
    }
}// End of class

File: app/src/main/java/com/csstudios/developers/theultimatestressreliever/AdviceActivity.kt
package com.csstudios.developers.theultimatestressreliever

import android.support.v7.app.AppCompatActivity
import android.content.Intent
import android.content.pm.ActivityInfo
import android.media.MediaPlayer
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import kotlinx.android.synthetic.main.activity_advice.*

class AdviceActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_advice)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT

        // Media players for advice (One for each recording)
        val adviceSound = MediaPlayer.create(this, R.raw.advice1)
        val adviceSound2 = MediaPlayer.create(this, R.raw.advice2)
        val adviceSound3 = MediaPlayer.create(this, R.raw.advice3)
        val adviceSound4 = MediaPlayer.create(this, R.raw.advice4)
        val adviceSound5 = MediaPlayer.create(this, R.raw.advice5)
        val adviceSound6 = MediaPlayer.create(this, R.raw.advice6)

        // Spinner content (Array from strings.xml)
        val adviceArray = resources.getStringArray(R.array.adviceArray)
        val adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, adviceArray)
        adapter.setDropDownViewResource(android.R.layout.simple_list_item_1)
        adviceSpinner.adapter = adapter

        // Execute spinner options to sounds
        adviceSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
                val selectedAdvice = parent?.selectedItem.toString()
                adviceButton.setOnClickListener {
                    when (selectedAdvice) {
                        "Fun Fact" -> adviceSound.start()
                        "Random Advice" -> adviceSound2.start()
                        "Life-changing Advice" -> adviceSound3.start()
                        "Another Fun Fact" -> adviceSound4.start()
                        "Did You Know?" -> adviceSound5.start()
                        "Just One More Thing..." -> adviceSound6.start()
                    }
                }
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                // Set TextView to "select an option" (strings.xml)"
                nothingSelected.text = resources.getString(R.string.nothingSelected)
            }
        }

        // Switch Activities
        activityButton2.setOnClickListener {
            val mainScreen = Intent(activityButton2.context, MainActivity::class.java)
            activityButton2.context.startActivity(mainScreen)
        }
    }
}